### ISSUE TITLE
Training loss not being tracked in multi gpu training with Pytorch Lightning

### ISSUE THREAD

**Issue by vikigenius** [aimhubio/aim#2897]
## 🐛 Bug

I am using Pytorch Lightning and noticing that the train loss is not being tracked when used with multiple gpus

### To reproduce

Run multi gpu training with Pytorch Lightning

Here are the log calls

```python
    def training_step(self, batch: Batch, batch_idx: int) -> Record:
        """Training Step."""
        outputs = self.forward(batch)
        loss = self.loss_module(outputs['query_embedding'], outputs['document_embedding'])
        self.log("train_loss", loss, on_epoch=True, sync_dist=True)
        return loss

    def validation_step(
        self,
        batch: Batch,
        batch_idx: int,
        dataloader_idx: int = 0
    ) -> torch.Tensor:
        """Validation Step."""
        outputs = self.forward(batch)
        loss = self.loss_module(outputs['query_embedding'], outputs['document_embedding'])
        self.log("val_loss", loss, on_epoch=True, sync_dist=True)
        self.metric.update(outputs['query_embedding'], outputs['document_embedding'])

    def on_validation_epoch_end(self):
        """Validation epoch end."""
        recalls = self.metric.compute()
        self.log_dict(recalls, sync_dist=True)
        self.metric.reset()
        self.metric.clear_embeddings()
```

### Expected behavior
I get most of the metrics and I do get the `loss (subset=val)` being plotted

But I do not get the `loss (subset=train)` at all.

I expect both the loss subsets to be plotted

### Environment

- Aim Version: 3.17.5
- Python version: 3.9
- pip version: 23.1.2
- OS: Linux

**Comment by vikigenius** [comment]
NVM, the issue stems from the fact that specifying `on_epoch` seems to restrict training loss reporting to just end of epoch.
